nxc/modules/pre2k.py

```diff
@@ -0,0 +1,150 @@
+            context.log.info(f"Connecting to LDAP server at ldap://{connection.host}")
+
+            if connection.kerberos:
+                ldap_connection = ldap.LDAPConnection(f"ldap://{connection.host}", connection.baseDN, None)
+                ldap_connection.kerberosLogin(connection.username, connection.password, connection.domain, lmhash=connection.lmhash, nthash=connection.nthash, aesKey=connection.aesKey, kdcHost=connection.kdcHost)
+            else:
+                ldap_connection = ldap.LDAPConnection(f"ldap://{connection.host}", connection.baseDN, None)
+                ldap_connection.login(connection.username, connection.password, connection.domain, lmhash=connection.lmhash, nthash=connection.nthash)
+
+            # Define the search filter for pre-created computer accounts
+            search_filter = "(&(objectClass=computer)(userAccountControl=4128))"
+            attributes = ["sAMAccountName", "userAccountControl", "dNSHostName"]
+
+            context.log.info(f"Using search filter: {search_filter}")
+            context.log.info(f"Attributes to retrieve: {attributes}")
+
+            computers = []
+
+            try:
+                # Use paged search to retrieve all computer accounts with specific flags
+                paged_search_control = ldapasn1.SimplePagedResultsControl(criticality=True, size=1000)
+                search_results = ldap_connection.search(searchFilter=search_filter, attributes=attributes, searchControls=[paged_search_control])
+
+                for item in search_results:
+                    if isinstance(item, ldapasn1.SearchResultEntry):
+                        context.log.debug(f"Raw item: {item.prettyPrint()}")
+
+                        sam_account_name = None
+                        user_account_control = None
+
+                        for attribute in item["attributes"]:
+                            context.log.debug(f"Attribute: {attribute.prettyPrint()}")
+                            if str(attribute["type"]) == "sAMAccountName":
+                                sam_account_name = str(attribute["vals"][0])
+                            elif str(attribute["type"]) == "userAccountControl":
+                                user_account_control = str(attribute["vals"][0])
+
+                        context.log.debug(f"Processing computer: {sam_account_name}, UAC: {user_account_control}")
+
+                        if sam_account_name and user_account_control is not None:
+                            user_account_control = int(user_account_control)
+
+                            # Check if the account is a pre-created computer account
+                            if user_account_control == 4128:  # 4096 | 32
+                                computers.append(sam_account_name)
+                                context.log.debug(f"Added computer: {sam_account_name}")
+
+                # Save computers to file
+                base_dir = "/root/.nxc/DiscoveredComputers"
```

Comment:
Please always use `NXC_PATH` as not everybody uses nxc as root. 
https://github.com/Pennyw0rth/NetExec/blob/3a5c109fedbcccda8895f134dfa9d078fc0cd8b2/nxc/paths.py#L5

And it would be nice if the output would be stored in `NXC_PATH/loot/pre2k/` or in `NXC_PATH/logs/pre2k/` depending on the output.

Reply:
Fixed